include/CriticalRSM.h

```diff
@@ -2,6 +2,7 @@
 #define CRITICAL_RSM_H
 
 #include "MatchingAlgorithm.h"
+#include <queue>
```

Comment:
why is a queue needed for the implementation here? the order of proposal shouldn't have any bearing on the matching output. if you see that the queue gives you correct results but the original freelist doesn't, that means there is a bug in the proposal driver.